hpx/util/pool_timer.hpp

```diff
@@ -0,0 +1,114 @@
+//  Copyright (c) 2016 Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_UTIL_POOL_TIMER)
+#define HPX_UTIL_POOL_TIMER
+
+#include <hpx/config.hpp>
+#include <hpx/util/function.hpp>
+#include <hpx/lcos/local/spinlock.hpp>
+
+#include <boost/thread/locks.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/enable_shared_from_this.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+
+namespace hpx { namespace util
+{
+    class pool_timer;
+}}
+
+namespace hpx { namespace util { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    class HPX_EXPORT pool_timer
+      : public boost::enable_shared_from_this<pool_timer>
+    {
+    private:
+        friend class util::pool_timer;
+
+        typedef lcos::local::spinlock mutex_type;
+
+    public:
+        pool_timer();
+        
+        pool_timer(util::function_nonser<bool()> const& f,
+            util::function_nonser<void()> const& on_term,
+            boost::posix_time::time_duration microsecs, std::string const& description,
```

Comment:
Prefer _chrono_ support to _posix_time_ support, so users can say nice things like `50ms`.

The facilities in [`date_time_chrono.hpp`](https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/date_time_chrono.hpp) allow you to support `boost::posix_time`, `boost::chrono`, and `std::chrono` with a single overload. This will required a specialized `basic_deadline_timer`, like the one used [here](https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/threads/detail/set_thread_state.hpp#L296-L300).